tests/test_finite_difference.py

```diff
@@ -8,79 +8,40 @@ def setup_method():
     np.random.seed(0)
 
 
-@pytest.mark.parametrize('x_data', [np.arange(10), 3])
-@pytest.mark.parametrize('init_model,init_par',
-    [
-        ({y: 3 * a * x**2}, {'a': 3.5}),
-        ({y: 3 * a * x**2 - sf.exp(b) * x},
-        {'a': 3.5, 'b': 2}),
-        ({y: 3 * a * x**2, z: sf.exp(a*x)}, {'a': 3.5}),
-        ({y: 3 * a * x**2 + b * x - c, z: sf.exp(a*x - b) * c},
-            {'a': 3.5, 'b': 2, 'c': 5})  
-    ]
-)
-def test_model(x_data, init_model, init_par):
+@pytest.mark.parametrize('x_data', [np.arange(10)/10, 3])
+@pytest.mark.parametrize('model,init_par',
+                         [
+                             (sf.Model({y: 3 * a * x**2}),
+                              {'a': 3.5}),
+                             (sf.Model({y: 3 * a * x**2 - sf.exp(b) * x}),
+                              {'a': 3.5, 'b': 2}),
+                             (sf.Model({y: 3 * a * x**2, z: sf.exp(a*x)}),
+                              {'a': 3.5}),
+                             (sf.Model({y: 3 * a * x**2 + b * x - c, z: sf.exp(a*x - b) * c}),
+                              {'a': 3.5, 'b': 2, 'c': 5}),
+                             (sf.Model({y: 3 * a * x**2 + b * x * w - c, z: sf.exp(a*x - b) + c*w}),
+                              {'a': 3.5, 'b': 2, 'c': 5, 'w': np.arange(10)/10}),
+                             (sf.Model({y: 3 * a * x**2 + b * x * w - c, z: sf.exp(a*x - b) + c*w}),
+                              {'a': 3.5, 'b': 2, 'c': 5, 'w': 5}),
+                             (sf.ODEModel({sf.D(y, x): -a * z, sf.D(z, x): y}, initial={y: 0, x: 1, z: 0}),
+                              {'a': 11})
+                         ]
```

Comment:
Please add comments to these cases on what behaviour they're testing. Especially add a comment on the places where you provide values for w.

Reply:
The short version of these comments is in the docstrings.

Reply:
It would still be nice to have those comments in the same place as the cases.